src/hooks/useForm.ts

```diff
@@ -0,0 +1,38 @@
+import React, { useCallback, useState } from 'react';
+
+interface StateType {
+  [index: string]: any;
+}
+
+const useForm = <T extends StateType>(initialStateObject: T) => {
+  const [obj, setObj] = useState<T>(initialStateObject);
+
+  const formHandler = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const newObj = { ...obj, [event.target.id]: event.target.value };
+      setObj((prev: T) => newObj);
+    },
+    [obj],
+  );
+
+  const selectOneHandler = useCallback(
+    (event: React.MouseEvent<any>, value: any) => {
```

Comment:
```suggestion
    (event: React.MouseEvent<HTMLElement>, value: any) => {
```
HTMLElement로 첫번째 any 지울 수 있고, value의 any는 타입스크립트의 제네릭을 이용하면 되는데요.
이건 리뷰로 설명하기는 너무 길어서 멘토링 시간에 알려드리겠습니다!

Reply:
이렇게 했을때 selectOneHandler 를 선언 후 처음 사용하는 부분에서 제네릭으로 value 타입을 지정해주면 되는거 맞을까요??

```tsx
// useForm 사용 컴포넌트 ( SignupPage )
...
const { obj: userInfo, formHandler, toggleHandler, selectOneHandler } = useForm(initalInfo);
...
return (
  ...
  <SelectType<StateType> selectOneHandler={selectOneHandler<boolean>} userInfo={userInfo} />
  ...
)
```

Reply:
```ts
const selectOneHandler = <A>(value: A) => {
  const newObj = { 123: value }

  return newObj
}

selectOneHandler<boolean>(true)
```